src/lib/keycard-pdf.ts

```diff
@@ -0,0 +1,227 @@
+import PDFDocument from 'pdfkit'
+import { Base64Encode } from 'base64-stream'
+import * as QRCode from 'qrcode'
+
+const QR_DATA_GAP = 15
+const QR_WIDTH = 128
+const DEFAULT_LOGO_PATH = '../../resources/sml-logo.png'
+
+const colors = {
+  REGULAR_COLOR: '#000',
+  TEXT_COLOR: '#3a3a3a',
+  INFO_COLOR: '#7a7a7a',
+  WARN_BG_COLOR: '#ffe4dd',
+  WARN_TEXT_COLOR: '#ff1c23'
+}
+
+const fonts = {
+  MONOSPACE_FONT: 'Monospace Font',
+  REGULAR_FONT: 'Regular Font',
+  STRONG_FONT: 'Strong Font'
+}
+
+const fontSizes = {
+  DOCUMENT_TITLE: 24,
+  HEADER_1: 20,
+  HEADER_3: 14,
+  HEADER_4: 12,
+  REGULAR: 10,
+  SMALL: 8
+}
+
+export const generatePdf = async (walletName: string, userPrivateKey: string, backupPrivateKey: string, servicePublicKey: string, logoPath?: string): Promise<string> => {
+  const doc = await generateBackupPdf(walletName, userPrivateKey, backupPrivateKey, servicePublicKey, logoPath)
+  const stream = doc.pipe(new Base64Encode())
+  let base64Pdf: string = ''
+  stream.on('data', (data: string) => base64Pdf += data)
+  return new Promise<string>((resolve: Function, reject: Function) => {
+    stream.on('end', () => resolve(base64Pdf))
+    stream.on('error', () => reject())
+  })
```

Comment:
Moze da sie to jakos zrobic bez mutowalnych zmiennych?

Reply:
W zasadzie dlaczego po prostu nie zwrocic streama?

Reply:
1. To już było wrzucone przez Piotrka
2. W internetach mówią, żeby tak robić - https://stackoverflow.com/a/39501140/1150503